nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/security/authorization/AuthorizerFactory.java

```diff
@@ -523,6 +523,9 @@ private Authorizer createDefaultAuthorizer() {
         return new Authorizer() {
             @Override
             public AuthorizationResult authorize(final AuthorizationRequest request) throws AuthorizationAccessException {
+                if(properties.isReadOnly() && !request.getAction().equals(RequestAction.READ)) {
```

Comment:
Injecting this property check into the default is not an optimal approach because it requires evaluating the property on every request, as opposed to configuring a specific authorizer.